Library/Homebrew/cmd/cleanup.rb

```diff
@@ -57,6 +57,7 @@ def cleanup_keg(keg)
   def cleanup_cache
     return unless HOMEBREW_CACHE.directory?
     HOMEBREW_CACHE.children.each do |path|
+      cleanup_path(path) { path.unlink } and next if path.fnmatch '*.incomplete'
```

Comment:
Not so well, as it turns out:
```
                              user     system      total        real
pathname.fnmatch          1.490000   0.150000   1.640000 (  1.644161)
pathname.to_s.end_with?   1.230000   0.160000   1.390000 (  1.382266)
```

Pushing changes now; thanks for the comments!